timelock/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+// This contract demonstrates 'timelock' concept and implements a
+// greatly simplified Claimable Balance (similar to
+// https://developers.stellar.org/docs/glossary/claimable-balance).
+// The contract allows to deposit some amount of token and allow another
+// account(s) claim it before or after provided time point.
+#![no_std]
+#[cfg(feature = "testutils")]
+extern crate std;
+
+use soroban_auth::{
+    check_auth, NonceAuth, {Identifier, Signature},
+};
+use soroban_sdk::{contractimpl, contracttype, BigInt, BytesN, Env, IntoVal, Symbol, Vec};
+use soroban_token_contract as token;
+use token::TokenClient;
```

Comment:
💡 We no longer need to have the token contract as a dependency outside of tests if you use the generated client. If you add the following you can then reference the client as `token::ContractClient`. This helps to illustrate how others might interact with contracts that aren't the token contract that don't have the hand coded machinery.

```suggestion
mod token {
    soroban_sdk::contractimport!(file = "../soroban_token_contract.wasm");
}
```

Reply:
Done, though I don't fully understand this. Will we somehow deduce that this is token contract and switch to the native version?

Reply:
You're already importing the client for the non-native contract and using it. For how this will work with the native contract, I think we'll probably package the specification into a file that can be imported here just like a wasm file, but it won't contain the program.'

Reply:
I see, thanks for the explanation.